numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -7,6 +7,19 @@
 #include "multiarraymodule.h"
 
 
+NPY_VISIBILITY_HIDDEN PyObject * npy_arrayfunction_str_like = NULL;
+NPY_VISIBILITY_HIDDEN PyObject * npy_arrayfunction_str_numpy = NULL;
+
+static int
+intern_strings(void)
+{
+    npy_arrayfunction_str_like = PyUString_InternFromString("like");
```

Comment:
If the first call here fails (returns `NULL`), the second should not be executed.

Reply:
Also, we should do something like this:

```C
if (npy_arrayfunction_str_like != NULL && npy_arrayfunction_str_numpy) {
    // Intern strings
}

// return strings
```

Threading isn't a concern since we're holding the GIL.

Reply:
> If the first call here fails (returns `NULL`), the second should not be executed.

I pretty much copied how it's done in https://github.com/numpy/numpy/blob/e3c52130add2ed5f7f0b9442b3b2742c09eb2de0/numpy/core/src/umath/umathmodule.c#L235-L259 . Is that unsafe?

Reply:
I believe so... The call can fail.

Reply:
While it seems logical that it can fail, the documentation doesn't mention that possibility: https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_InternFromString . `PyUString_InternFromString` is a macro to that: https://github.com/numpy/numpy/blob/e3c52130add2ed5f7f0b9442b3b2742c09eb2de0/numpy/core/include/numpy/npy_3kcompat.h#L131 .

With that said, I'd be inclined to just keep on mimicking the same behavior as already done in the rest of NumPy. Any objections?